### PR TITLE
Add black-doc to pre-commit

### PR DESCRIPTION
**Pull request by akaszynski**
Redo of #4119 whereby only black doc is added. Style check is expected to fail. We will have a follow up PR implementing the style changes.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,6 +4,13 @@ repos:
   hooks:
   - id: black
 
+- repo: https://github.com/keewis/blackdoc
+   rev: v0.3.8
+   hooks:
+   - id: blackdoc
+     files: '\.py$'
+     exclude: pyvista/plotting/charts.py
```
Comment (adeak) [inline .pre-commit-config.yaml:12]: What's wrong with charts?

Comment (akaszynski) [reply .pre-commit-config.yaml]: The use of `docsubs`:

https://github.com/pyvista/pyvista/blob/a26a52d037a265774be0ca06bb141e3346e73ad0/pyvista/plotting/charts.py#L1773-L1781

I can’t think of a workaround for this. We may consider opening an issue on https://github.com/keewis/blackdoc and seeing if we can implement a skip.

**Review by tkoyama010 — APPROVED** [review]

**Review by adeak — CHANGES_REQUESTED** [review]
Let's hold back on this until any issues regarding https://github.com/pyvista/pyvista/pull/4119 are settled and git ignore revs items are added here.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (akaszynski) [reply .pre-commit-config.yaml]: Issue opened:
https://github.com/keewis/blackdoc/issues/172

**Comment by akaszynski** [comment]
Reminder to add:
```

[tool.blackdoc]
# From https://numpydoc.readthedocs.io/en/latest/format.html
# The length of docstring lines should be kept to 75 characters to facilitate
# reading the docstrings in text terminals.
line-length = 72
```

to `pyproject.toml`.

**Comment by akaszynski** [comment]
> Let's hold back on this until any issues regarding #4119 are settled and git blame ignore revs items are added here.

Changes addressed in #4125

**Comment by tkoyama010** [comment]
@adeak Could you approve this? I cannot wait using black-doc :slightly_smiling_face:

**Comment by akaszynski** [comment]
Should be good to go @adeak.

File: .git-blame-ignore-revs [hunk 2]
```diff
@@ -1,3 +1,6 @@
 # enforce unified code style with black
 28914fc2236cd5850f536b151d0ef383855d8519
 db0691cfb6b033e55664ffba25deb456a386a48c
+
+# enforce unified code in our documentation examples style with blackdoc
```
Comment (akaszynski) [inline .git-blame-ignore-revs:5]:
```suggestion
# enforce unified code style in our documentation with blackdoc
```

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -1,3 +1,6 @@
 # enforce unified code style with black
 28914fc2236cd5850f536b151d0ef383855d8519
 db0691cfb6b033e55664ffba25deb456a386a48c
+
+# enforce unified code style in our documentation with blackdoc
+a26a52d037a265774be0ca06bb141e3346e73ad0
```
Comment (adeak) [inline .git-blame-ignore-revs:6]: I think the conclusion in https://github.com/pyvista/pyvista/pull/4129 was to ignore that too (basically one PR broken in two due to a late reconfiguration of blackdoc):
```suggestion
a26a52d037a265774be0ca06bb141e3346e73ad0
635052d1bce8615fcc4879f5f200921a4f78aa1a
```
Or did I misunderstand?

Comment (akaszynski) [reply .git-blame-ignore-revs:6]: You're correct. Thanks for catching that!

**Review by adeak — APPROVED** [review]

**Pull request merged**